cfn-resources/markdown-generator/generate_version_markdown.go

```diff
@@ -0,0 +1,137 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+)
+
+const mongoDBPublisherID = "bb989456c78c398a858fef18f2ca1bfc1fbba082"
+const ouputMarkdownFile = "resource-versions.md"
+
+func main() {
+	// can potentially get regions from github publish yaml file
+	var regions = []string{"af-south-1", "ap-east-1", "ap-northeast-1", "ap-northeast-2", "ap-northeast-3", "ap-south-1", "ap-southeast-1", "ap-southeast-2", "ap-southeast-3",
+		"ca-central-1", "eu-central-1", "eu-north-1", "eu-south-1", "eu-west-1", "eu-west-2", "eu-west-3", "me-south-1", "sa-east-1", "us-east-1", "us-east-2", "us-west-1", "us-west-2",
+		"ap-south-2", "ap-southeast-4", "eu-central-2", "eu-south-2", "me-central-1"}
+
+	var resourceTypes = GetTypeNamesFromRpdkJSONFiles()
+
+	// map of resource type name to list of values where each index corresponds to a region
+	resourceVersions := obtainVersions(regions, resourceTypes)
+	generateMarkdown(ouputMarkdownFile, regions, resourceTypes, resourceVersions)
+}
+
+func obtainVersions(regions, resourceTypes []string) map[string][]string {
+	resourceValues := map[string][]string{}
+
+	for _, typeName := range resourceTypes {
+		resourceValues[typeName] = make([]string, len(regions))
+	}
+
+	for i, region := range regions {
+		typeSummaries := getTypeSummariesOfRegion(region)
+
+		for _, typeSummary := range typeSummaries {
+			fmt.Printf("Region: %s, type name: %s, version: %s, date: %s\n", region, *typeSummary.TypeName, *typeSummary.LatestPublicVersion, typeSummary.LastUpdated.String())
+			if _, exists := resourceValues[*typeSummary.TypeName]; exists {
+				if typeSummary.LatestPublicVersion != nil && typeSummary.LastUpdated != nil {
+					date := *typeSummary.LastUpdated
+					resourceValues[*typeSummary.TypeName][i] = fmt.Sprintf("%s (%s)", *typeSummary.LatestPublicVersion, date.Format("2006-01-02"))
+				}
+			}
+		}
+	}
+	return resourceValues
+}
+
+func generateMarkdown(filename string, regions, resourceTypes []string, resourceValues map[string][]string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	for i, r := range regions {
+		regions[i] = useNoBreakChar(r)
+	}
+
+	headers := []string{"Latest version"}
```

Comment:
knitpit: I think this is great.  for a newer version we may want to use Go templates, e.g. this is the one used by AWS to generate the model.go files: https://github.com/aws-cloudformation/cloudformation-cli-go-plugin/blob/fc208d1da48a82f84cc3741b71bf5e467f0e07ec/python/rpdk/go/templates/types.go.tple

Reply:
This is a great point, did not have it mind during implementation. if we continue to invest here, definitely worth refactoring for better maintainability of the markdown strucuture.